param_sync.py

```diff
@@ -7,12 +7,39 @@ class ParamSyncRule:
     training.
     """
 
+    def update_core(self, local_params, master_params):
+        """
+        Iterator over updates for local and master params.
+
+        This function must be ready to deal with symbolic or numerical
+        inputs.
```

Comment:
why it need to handle numerical inputs? Currently it is only used with symbolic variable.